### DISCUSSION TITLE
Importing Posts to a new installation

### DISCUSSION THREAD

**Discussion by Auroraboreali-s** [flyntwp/flynt#396] [Q&A] [answered]
Hi guys, 

How do you go about importing blog posts to a new installation. Usually you can use the Wordpress import/export tool, export some specific category and then import this in a new installation. However, since Flynt is component based, every post doesn't have any text area selected as standard, the only thing you can import is the title of the posts. 

Let's assume all of your posts are text based only. 

Is there any way to have the wysiwyg block component enabled temporarily as standard for posts? so you could import the content of the posts too?

What would be way to import posts like that?

**Comment by aaronmeder** [comment, marked as answer]
Hey @Auroraboreali-s 

not sure if I understood you correctly: you have existing posts that used the standard Wordpress editor or Gutenberg and want to keep them on the new site, correct?

I had a similar setup where a blog had 1000+ existing posts they wanted to keep (and are great for SEO of course) but they wanted to compose the new blog posts using Flynt components.

**Importing the posts**
Importing the posts with the media was quite a hassle to be honest.
Easiest would be if you can start on a copy of the existing WP site and just enable Flynt as a new theme, then you don't need to import anything.
I've used the [Export media with selected content plugin](https://de.wordpress.org/plugins/export-media-with-selected-content/) addon for the stanard WP export and exported posts (with media) by year - otherwise it would result in timeouts while importing. 


**Displaying the posts**
I've adjusted the "post-main" section of templates/single.twig a little bit:
```php
<section class="post-main">
	{# Display the classic WP content field on older posts #}
	{% if (post.date('U')) <= '2021-07-05'|date('U') %}
		<div class="classic-wp-content container centerMaxWidthContainer">
			<div class="content centerContentMaxWidth">
				{{ post.content }}
			</div>
		</div>
	{% endif %}

	{# ACF Components #}
	{% for component in post.meta('postComponents') %}
		{{ renderComponent(component) }}
	{% endfor %}
</section>
```

I've a blog post is older than the given date it will render the normal wordpress content field for it.
Then in inc/removeEditor.php i've commented out line 11:
```php
//remove_post_type_support('post', 'editor');
```
This will make the wordpress editor to show up again in the admin area.

Hope that helps!
Aaron

**Reply by Auroraboreali-s** [reply]
Hi @aaronmeder

This is actually pretty brilliant solution, especially if your posts are plain text. I can imagine if you have lots of images the newest Gutenberg editor can make it look messy so you would need to fix it here and there.

The posts that need to be updated, they can be manually converted to a component structure.

Perfect.

Changing how the older posts render based on the date in the single template just gets the content from the database and it's all available in de website.

Great stuff! Thank you!

**Reply by aaronmeder** [reply]
@Auroraboreali-s Glad I could be of help!

If this provided what you needed I think you should be able to mark the discussion as "answered".

Happy evening.
Aaron
